### ISSUE TITLE
How do I merge this with  CodeReclaimers/neat-python master?

### ISSUE THREAD

**Issue by nexon33** [bennr01/neat-python#10]
I feel like a complete noob for asking, but after almost a whole day of trying and still failing to figure this out I figure I'd just ask.

**Comment by bennr01** [comment]
It's been a long time since I've worked on this repository, but the rough process should look like this (assuming you want the updates for `neat.distributed`):

**Assuming you are using git CLI:**

1. clone CodeReclaimers's repository (`git clone https://github.com/CodeReclaimers/neat-python.git`). If you already have a clone, you can use that. Beware that merges involving merge conflicts can easily mess up your code if you are inexperienced, so either use a separate branch or make a backup if you've made any changes to your clone/fork.
2. inside the repository directory, add this repo (`git remote add bennr01 git@github.com:bennr01/neat-python.git`)
3. checkout the branch you want to merge into (`git checkout master`)
4. fetch and merge the changes (either `git pull bennr01 distributed_socket` or `git fetch bennr01 && git merge bennr01/distributed_socket`).
5. As this will result in a merge conflict, you'll need to resolve that one. I believe it should be enough to take all changes to `neat/distributed.py` and discard all changes to `.travis.yml`, but I'm not sure. Once you've resolved all conflicts (beware: text files will be modifed, be sure that you've removed the various markers in the file), commit the changes `git commit` (then enter a message, save and quit).
6. Done.

**The quick & dirty way**

If you don't care about the correct git history or that the tests may fail, you can simply copy the `neat/distributed.py` file from this repository ([raw version here](https://raw.githubusercontent.com/bennr01/neat-python/distributed_socket/neat/distributed.py)) and replace it in the base repository. You'll need to change any use of `authkey` in the example to unicode.

**Comment by nexon33** [comment]
Okay thanks! it does seem to work, I seperated the primary and secondary clients in 2 files. I start both and when I quit the secondary client then it throws an error in the primary client instead of trying to reconnect (the master code does this, but only sometimes succesfully). I'm currently trying to fix that somehow but I'm not really confident I'll be able to.

**Comment by nexon33** [comment]
Any chance you can fix this for me? I can offer you some money if you manage to get NEAT running in a way that allows the clients to lose connection and be able to always reconnect. Not a lot of money but money none the less

**Comment by bennr01** [comment]
I can take a look once I've got some time, but you don't have to offer me any money for this. Simply asking would have been enough. BTW, have you set `reconnect=True` in the `.start()` method?

**Comment by nexon33** [comment]
Yes I did try that already, I'll check it out tomorrow again im going to look through the what must already be 30 cloned repos I editted. I did also get the latest version running without socks but with reconnects (only sometimes visible) 

The other times it didn't reconnect but I could see a seperate thread being spawned and removed about once a second.

**Comment by nexon33** [comment]
okay so it is running [this](https://github.com/CodeReclaimers/neat-python/blob/master/tests/test_xor_example_distributed.py) in linux which works, I commented out the secondary processes from the file running on linux (WSL) and then commented out the primary process on windows and with only the secondary process it does what it has to do without any weird errors.

the current behaviour is almost as desired except for the fact that when I stop debugging the windows process (secondary process) it only sometimes reconnects and sometimes it kind of reconnects (I can see threads being generated in the debugger of the primary script) but there is no useful processing being done anymore.

edit: I've also narrowed down that the problem is somewhere in the primary node, I do not know where however

**Comment by nexon33** [comment]
any eta on when you'd pick this up? If its gonna be a while I should probably look for another solution.
